cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -54,23 +54,23 @@ type SpanHandlerBuilder interface {
 }
 
 // NewSpanHandlerBuilder returns a span handler
-func NewSpanHandlerBuilder(opts ...basicB.Option) (SpanHandlerBuilder, error) {
+func NewSpanHandlerBuilder(cOps *CollectorOptions, flagOpts *flags.Flags, opts ...basicB.Option) (SpanHandlerBuilder, error) {
```

Comment:
I think we can simplify this a lot.
  * currently SpanHandlerBuilder is an interface. Can we make it a struct instead, and do the dispatching by storage type inside the `BuildHandlers()`?
  * if so, then I think we can merge CollectorOptions into SpanHandlerBuilder, similar to the pattern used in AgentBuilder
  * a lot of functions in this file can be methods on the SpanHandlerBuilder, simplifying the signatures